packages/grid/x-data-grid/src/internals/hooks/features/columns/gridColumnsUtils.test.tsx

```diff
@@ -0,0 +1,139 @@
+import { expect } from 'chai';
+import { computeFlexColumnsWidth } from './gridColumnsUtils';
+
+describe('gridColumnsUtils', () => {
+  describe('computeFlexColumnsWidth', () => {
```

Comment:
Feel free to suggest test scenarios if you have some in mind :)

Reply:
Maybe a test with 2 columns having max width, one being limited by its max width and the other not.
For instance :

1000px to allocate
Col1 has maxWidth 400
Col2 has maxWidth 800

=> output 400 / 600

Reply:
I think it's already there - https://github.com/mui/mui-x/pull/4006/files#diff-877ed0a1b46c1358b40fa12ad241ccda10b0421925abcc52f9c1aac5fba85baaR102
Unless you meant something else?

Reply:
Here only one of the two columns has a maxWidth